starsessions/stores/redis_store.py

```diff
@@ -38,13 +46,16 @@ def __init__(
 
         self.gc_ttl = gc_ttl
         self.prefix: typing.Callable[[str], str] = prefix
-        self._connection: aioredis.Redis = connection or aioredis.from_url(url)  # type: ignore[no-untyped-call]
+        if connection:
+            self._connection: BaseRedis = connection
+        elif url:
+            self._connection = from_url(url)
```

Comment:
The previous solution always sets the connection. This change makes  connection optional because it doesn't handle "else" case. Please replace "elif" with "else". We check configuration in the code above.

Reply:
yep sorry that was bad

Reply:
well I now know why I did this to please mypy, is the assert ok for you ?

Reply:
Yes, totally. Just make sure it always truthy.